pkg/controller/operators/olm/envvar/inject.go

```diff
@@ -0,0 +1,61 @@
+package envvar
+
+import (
+	"errors"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// InjectEnvIntoDeployment injects the proxy env variables specified in
+// proxyEnvVar into the container(s) of the given PodSpec.
+//
+// If any Container in PodSpec already defines an env variable of the same name
+// as any of the proxy env variables then it
+func InjectEnvIntoDeployment(podSpec *corev1.PodSpec, envVars []corev1.EnvVar) error {
+	if podSpec == nil {
+		return errors.New("no pod spec provided")
+	}
+
+	for i := range podSpec.Containers {
+		container := &podSpec.Containers[i]
+		container.Env = merge(container.Env, envVars)
+	}
+
+	return nil
+}
+
+func merge(containerEnvVars []corev1.EnvVar, newEnvVars []corev1.EnvVar) (merged []corev1.EnvVar) {
+	merged = containerEnvVars
+
```

Comment:
Thoughts?
- If an `env` variable that is being injected is already defined in the `csv` deployment spec, it will be overwritten.
- For proxy env var, empty means unset and will not result in an env var. Can we say the same for all other env variable(s) in pod configuration?

Reply:
yes, I agree on allowing the user to overwrite a "set" env var to ""
proxy env is an exception, if they are empty they are not injected. and if the use specifies any in subscription pod ( empty or not ) it overrides the proxy, we won't even query `cluster` proxy at that point.

Reply:
Given this, we need to treat the proxy env differently, I will add that change ( maybe in the followup PR I am planning if I don't have enough time )